xla/stream_executor/cuda/cuda_dnn.cc

```diff
@@ -4578,6 +4582,322 @@ void FixDimsForRaggedOffset(std::vector<int64_t>& dims, int max_reg_per_batch) {
   dims[0] *= max_reg_per_batch;
 }
 
+class ScoreModFunc {
```

Comment:
Can you put this class into its own file and add some tests? A class level comment would also be great.

Reply:
Done.